exec/java-exec/src/main/java/org/apache/drill/exec/service/ServiceEngine.java

```diff
@@ -142,7 +142,13 @@ private static BufferAllocator newAllocator(
 
   public DrillbitEndpoint start() throws DrillbitStartupException, UnknownHostException{
     int userPort = userServer.bind(config.getInt(ExecConstants.INITIAL_USER_PORT), allowPortHunting);
-    String address = useIP ?  InetAddress.getLocalHost().getHostAddress() : InetAddress.getLocalHost().getCanonicalHostName();
+    String configIP = config.getString(ExecConstants.BIT_ADVERTISED_HOST);
+    String address = null;
+    if (configIP == "") {
```

Comment:
You could use `config.hasPath` instead of relying on matching on an empty string.